src/main/java/seedu/address/ui/UpcomingTuitionListPanel.java

```diff
@@ -33,24 +35,20 @@
     public UpcomingTuitionListPanel(ObservableList<Student> studentList) {
         super(FXML);
         ObservableList<Tuition> tuitionList = FXCollections.observableArrayList();
-        ObservableList<Tuition> finalTuitionList = FXCollections.observableArrayList();
 
         populateTuitionList(studentList, tuitionList);
-        filterOneWeekTuitionSessions(tuitionList, finalTuitionList);
-        sortByDate(finalTuitionList);
+        sortByDate(tuitionList);
 
         studentList.addListener((ListChangeListener<Student>) change -> {
             while (change.next()) {
                 tuitionList.clear();
-                finalTuitionList.clear();
                 populateTuitionList(studentList, tuitionList);
-                filterOneWeekTuitionSessions(tuitionList, finalTuitionList);
-                sortByDate(finalTuitionList);
-                populateUpcomingTuitionListView(finalTuitionList);
+                sortByDate(tuitionList);
+                populateUpcomingTuitionListView(tuitionList);
             }
         });
 
-        populateUpcomingTuitionListView(finalTuitionList);
+        populateUpcomingTuitionListView(tuitionList);
```

Comment:
Would suggest to abstract:
```
populateTuitionList(studentList, tuitionList);
sortByDate(tuitionList);
populateUpcomingTuitionListView(tuitionList);
```
into another function. seems like you're repeating these 3 lines both outside and inside the listener.